src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -602,6 +603,15 @@ protected <T> SearchDocumentResponse.EntityCreator<T> getEntityCreator(ReadDocum
 	// region Entity callbacks
 	protected <T> T maybeCallbackBeforeConvert(T entity, IndexCoordinates index) {
 
+		// get entity metadata
```

Comment:
This would be called for every conversion of every entity, so that if the user saves 100 entities we have 100 calls.

Reply:
Yes, until I find a better place to cache/load metadata 😄